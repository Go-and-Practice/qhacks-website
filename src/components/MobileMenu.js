import React from "react";
import hamburger from "../assets/img/hamburger.svg";
import menuClose from "../assets/img/menuClose.svg";
import logo from "../assets/img/whiteLogo.svg";
import word from "../assets/img/whiteWordmark.svg";
import cubes from "../assets/img/mobileMenuCubes.png";

const MobileMenu = (props) => (
  <div
    align="right"
    css={{
      marginLeft: "auto",
      padding: "32px",
      "@media(min-width: 820px)": {
        display: "none"
      }
    }}
  >
    <img
      onClick={props.toggleMenu}
      src={hamburger}
      css={{ width: "32px", height: "24px" }}
    />
    <div
      className={props.isMenuVisible ? "no-scroll" : ""}
      css={{
        display: props.isMenuVisible ? "block" : "none",
        position: "fixed",
        width: "100vw",
        height: "100vh",
        overflow: "hidden",
        top: "0px",
        left: "0px",
        zIndex: "5",
        background: "#11213f"
      }}
    >
      <img
        align="right"
        css={{ padding: "32px" }}
        src={menuClose}
        onClick={props.toggleMenu}
      />
      <div
        css={{
          marginTop: "100px",
          color: "white",
          display: "grid",
          gridTemplateColumns: "1fr"
        }}
      >
        <div
          css={{
            marginLeft: "auto",
            marginRight: "auto",
            textAlign: "center",
            " a": {
              color: "white",
              textTransform: "uppercase",
              ":hover": {
                color: "#fedb01"
              }
            }
          }}
        >
          <div>
            <img src={logo} css={{ paddingBottom: "11px" }} />
          </div>
          <div css={{ paddingBottom: "52px" }}>
            <img src={word} width="134px" height="34.8px" />
          </div>
          {props.menuItems.map((i) => (
            <div key={i} css={{ paddingBottom: "24px" }}>
              <a href={`#${i}`} onClick={props.toggleMenu}>
                {i}
              </a>
            </div>
          ))}
          <div>
            <img src={cubes} />
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default MobileMenu;
