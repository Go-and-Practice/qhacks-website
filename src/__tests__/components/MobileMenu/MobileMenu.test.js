import React from "react";
import MobileMenu from "../../../components/MobileMenu";
import { shallow } from "enzyme";

describe("MobileMenu", () => {
  it("renders correctly", () => {
    expect(
      shallow(<MobileMenu breadCrumbs={["I", "LOVE", "SnApShOtS"]} />)
    ).toMatchSnapshot();
  });
});
