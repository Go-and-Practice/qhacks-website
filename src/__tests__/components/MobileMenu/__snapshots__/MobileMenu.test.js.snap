// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu renders correctly 1`] = `
<div
  align="right"
  css={
    Object {
      "@media(min-width: 820px)": Object {
        "display": "none",
      },
      "marginLeft": "auto",
      "padding": "32px",
    }
  }
>
  <img
    css={
      Object {
        "height": "24px",
        "width": "32px",
      }
    }
    src="test-file-stub"
  />
  <div
    className=""
    css={
      Object {
        "background": "#11213f",
        "display": "none",
        "height": "100vh",
        "left": "0px",
        "overflow": "hidden",
        "position": "fixed",
        "top": "0px",
        "width": "100vw",
        "zIndex": "4",
      }
    }
  >
    <img
      align="right"
      css={
        Object {
          "padding": "32px",
        }
      }
      src="test-file-stub"
    />
    <div
      css={
        Object {
          "color": "white",
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "marginTop": "100px",
        }
      }
    >
      <div
        css={
          Object {
            " a": Object {
              ":hover": Object {
                "color": "#fedb01",
              },
              "color": "white",
              "textTransform": "uppercase",
            },
            "marginLeft": "auto",
            "marginRight": "auto",
            "textAlign": "center",
          }
        }
      >
        <div>
          <img
            css={
              Object {
                "paddingBottom": "11px",
              }
            }
            src="test-file-stub"
          />
        </div>
        <div
          css={
            Object {
              "paddingBottom": "52px",
            }
          }
        >
          <img
            height="34.8px"
            src="test-file-stub"
            width="134px"
          />
        </div>
        <div
          css={
            Object {
              "paddingBottom": "24px",
            }
          }
          key="I"
        >
          <a
            href="#I"
          >
            I
          </a>
        </div>
        <div
          css={
            Object {
              "paddingBottom": "24px",
            }
          }
          key="LOVE"
        >
          <a
            href="#LOVE"
          >
            LOVE
          </a>
        </div>
        <div
          css={
            Object {
              "paddingBottom": "24px",
            }
          }
          key="SnApShOtS"
        >
          <a
            href="#SnApShOtS"
          >
            SnApShOtS
          </a>
        </div>
        <div>
          <img
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
